### ISSUE TITLE
Broadcasting cholesky

### ISSUE THREAD

**Issue by TomMaullin** [dask/dask#5371]
I am currently writing some code that needs to do many cholesky factorizations;

e.g. I wish to do `da.linalg.cholesky(M)` where `M` is actually a 3D matrix and I want to broadcast the operation across the first dimension so that for each `i` the cholesky of M[i,:,:] is calculated and the output `L` is stored as `L[i,:,:]`; much like how this is done in numpy's linalg.

I have played around with it a bit and am under the current impression this broadcasting is not yet supported in dask,.... if I am mistaken please let me know but I have tried for a while. If this feature is not present I would like to make a feature request as this is an increasingly in demand operation and is especially prominent in a lot of statistical estimation methods at the moment.

**Comment by shoyer** [comment]
Dask array generally just wraps NumPy/SciPy routines. As far as I can tell, SciPy doesn't include a broadcasting cholesky. So it's probably out of scope for Dask.

I would suggest writing your own wrapper for this, e.g,. by chunking your array into size 1 chunks along the leading dimension, and making use of Dask's `map_blocks` to calculate the operations in parallel.

If your arrays are small, then you might really need batching for efficiency. In that case, you might try wrapping the cholesky operation from TensorFlow (which does support batching) in `map_blocks`.

**Comment by TomMaullin** [comment]
Hmm I see... I will attempt this. Thank you.

**Comment by TomAugspurger** [comment]
Closing this for now, since scipy doesn't have a broadcasting cholesky. We can revisit if / when that's implemented.
